File: tck/scala-tck/src/main/scala/kalix/tck/model/valueentity/ValueEntityConfiguredEntity.scala
/*
 * Copyright (C) 2021-2024 Lightbend Inc. <https://www.lightbend.com>
 */

package kalix.tck.model.valueentity

import kalix.scalasdk.valueentity.ValueEntity
import kalix.scalasdk.valueentity.ValueEntityContext

/** A value entity. */
class ValueEntityConfiguredEntity(context: ValueEntityContext) extends AbstractValueEntityConfiguredEntity {
  override def emptyState: Persisted = Persisted.defaultInstance

  override def call(currentState: Persisted, request: Request): ValueEntity.Effect[Response] =
    effects.reply(Response.defaultInstance)
}

File: tck/java-tck/src/main/java/kalix/tck/model/valueentity/ValueEntityConfiguredEntity.java
/*
 * Copyright (C) 2021-2024 Lightbend Inc. <https://www.lightbend.com>
 */

package kalix.tck.model.valueentity;

import kalix.javasdk.valueentity.ValueEntityContext;
import kalix.tck.model.valueentity.ValueEntityApi.*;

public class ValueEntityConfiguredEntity extends AbstractValueEntityConfiguredEntity {

  public ValueEntityConfiguredEntity(ValueEntityContext context) {}

  public Effect<Response> call(Persisted state, Request request) {
    return effects().reply(Response.getDefaultInstance());
  }

  @Override
  public Persisted emptyState() {
    return null;
  }
}
